shared/settings/notifications/index.js

```diff
@@ -1,41 +1,62 @@
 // @flow
 import React from 'react'
-import {Box, Button, Checkbox, HeaderHoc, ProgressIndicator, Text} from '../../common-adapters'
-import {isMobile} from '../../constants/platform'
+import {Box, Button, Checkbox, ProgressIndicator, Text} from '../../common-adapters'
 import {globalStyles, globalMargins} from '../../styles'
 
 import type {Props} from './index'
 
+const makeCheckbox = (
+  group: string,
+  s: {name: string, description: string, subscribed: boolean},
+  props: Props
+) => (
+  <Checkbox
+    style={{marginTop: globalMargins.small, marginRight: globalMargins.medium}}
+    key={s.name}
+    disabled={!props.allowEdit}
+    onCheck={() => props.onToggle(group, s.name)}
+    checked={s.subscribed}
+    label={s.description}
+  />
+)
+
 const Notifications = (props: Props) =>
-  !props.settings
+  !props.groups.email
     ? <Box style={{...globalStyles.flexBoxColumn, flex: 1, justifyContent: 'center', alignItems: 'center'}}>
         <ProgressIndicator type="Small" style={{width: globalMargins.medium}} />
       </Box>
     : <Box style={{...globalStyles.flexBoxColumn, padding: globalMargins.small, flex: 1}}>
         <Text type="BodyBig" style={{marginTop: globalMargins.medium}}>Email me:</Text>
         <Box style={globalStyles.flexBoxColumn}>
-          {!!props.settings &&
-            props.settings.map(s => (
-              <Checkbox
-                style={{marginTop: globalMargins.small}}
-                key={s.name}
-                disabled={!props.allowEdit}
-                onCheck={() => props.onToggle(s.name)}
-                checked={s.subscribed}
-                label={s.description}
-              />
-            ))}
+          {props.groups.email.settings &&
+            props.groups.email.settings.map(s => makeCheckbox('email', s, props))}
         </Box>
         <Text type="BodyBig" style={{marginTop: globalMargins.medium}}>Or:</Text>
         <Checkbox
-          style={{marginTop: globalMargins.small, marginBottom: globalMargins.medium}}
-          onCheck={() => props.onToggleUnsubscribeAll()}
+          style={{marginTop: globalMargins.small}}
+          onCheck={() => props.onToggleUnsubscribeAll('email')}
           disabled={!props.allowEdit}
-          checked={!!props.unsubscribedFromAll}
+          checked={props.groups.email && !!props.groups.email.unsubscribedFromAll}
           label="Unsubscribe me from all mail"
         />
+
+        {!!props.groups.app_push &&
+          props.groups.app_push.settings &&
+          <Box style={globalStyles.flexBoxColumn}>
```

Comment:
we should make a helper for this or something as its really copy pasta of the above right?